src/main/java/org/opensearch/plugin/insights/rules/model/SearchQueryRecord.java

```diff
@@ -154,6 +155,7 @@ public SearchQueryRecord(final StreamInput in) throws IOException, ClassCastExce
             });
         }
         this.attributes = Attribute.readAttributeMap(in);
+        this.isCancelled = false;
```

Comment:
I think we need to read the boolean from the input stream and add a version check:
```
if (in.getVersion().onOrAfter(Version.V_3_1_0)) {
    this.isCancelled = in.readBoolean();
} else {
    this.isCancelled = false;
}
```
Also update the [writeTo](https://github.com/opensearch-project/query-insights/blob/bcae2c97c63e218789365312cca1bf8726ba3bdc/src/main/java/org/opensearch/plugin/insights/rules/model/SearchQueryRecord.java#L524) method in a similar fashion unless this info is not needed in node-to-node communication?